src/libmongoc/src/mongoc/mongoc-topology.c

```diff
@@ -602,6 +621,12 @@ mongoc_topology_new (const mongoc_uri_t *uri, bool single_threaded)
 
       mongoc_topology_description_add_server (td, elem->host_and_port, &id);
       mongoc_topology_scanner_add (topology->scanner, elem, id, false);
+
+      if (!found_nongenuine_host &&
+          (found_nongenuine_host = _is_nongenuine_host (elem->host))) {
+         /* TODO: the final copy for this log message is TBD */
+         MONGOC_INFO ("Nongenuine host detected");
```

Comment:
@alexbevi: I just noticed that [WEBSITE-15470](https://jira.mongodb.org/browse/WEBSITE-15470) mentions separate pages for Cosmos and DocumentDB. Do I assume correctly that we'll need to log distinct messages for each of those platforms?

That was present in my original draft of this PR but I ended up removing it when moving the check to `mongoc_topology_new()` (for no particular reason other than it made the implementation simpler). I can restore separate log messages if needed, though.

Reply:
@jmikola that is correct (distinct URLs will be shown based on host detection)

Reply:
* CosmosDB: https://www.mongodb.com/supportability/cosmosdb
* DocumentDB: https://www.mongodb.com/supportability/documentdb

Reply:
These URLs are now incorporated into the PR. I opted to omit the trailing period from the log message to avoid users copy-pasting the URL with a period and having it 404.